### ISSUE TITLE
	 * +G - softcallerid - Server Side Ignore for users not on your channels 	 * +g - callerid     - Server Side Ignore (for privmsgs etc)

### ISSUE THREAD

**Issue by gh0st68** [ValwareIRC/valware-unrealircd-mods#8]
Request by: gh0st 

**Please describe, in full, the functions and features of the module you would like to request**

In IRCd, the +g user mode, often referred to as "Caller ID" mode, is a potent feature. Activating this mode enables a server-side ignore function. Consequently, any user who sends a message to a client operating in +g mode will receive a numeric notification informing them that the recipient is in +g mode. The recipient, the +g mode client, will also get an alert that someone has messaged them while they're in +g mode. This alert is sent to the +g client once per minute to prevent message flooding, but there is no such rate limit for the sender, who will receive a numeric message with each message sent.

This mode also has a variation, G (uppercase G), which only blocks messages from users who are not in the same channels as you are.

The ACCEPT command is used to allow messages from specific users while in +g mode. This command accepts comma-separated parameters of users to be added or removed, in the format 'nick!user@host'. If only the nick is specified, the IRCD will convert it to 'nick!@'. Errors are returned if the accept list is full, if a user tries to add a mask to their accept list that already exists, or if a user tries to remove a mask from their accept list that is not there.

When a user is accepted, your client will continue receiving messages from them until they are manually removed from your accept list, you quit, or they change their nick (if the mask is based only on the nick).

You can view the accept list by issuing the 'ACCEPT *' or 'ACCEPT' command without parameters. To remove a mask from your accept list, issue 'ACCEPT -<mask>'. It is also possible to add and remove masks simultaneously.

**If you are requesting a feature that exists in another IRC project, please link to it below**

Inspircd, Hybrid, Ratbox

**Describe alternatives you've considered**

NOTHING!

**Additional context**

- gh0st

**Comment by TehPeGaSuS** [comment]
I agree that we could use something similar to how the other IRCd do it.

Unfortunately, user mode `+G` is already taken: `G = Filters out all Bad words in your messages with <censored>`.

IMHO, only `+g` would be a good starting point.

Cheers!
